components/datetime/src/helpers.rs

```diff
@@ -0,0 +1,36 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+/// Generates a test that checks the stack size of an item and a macro
+/// that should be used with `#[doc]` to document it.
+///
+/// The size should correspond to the Rust version in rust-toolchain.toml.
+///
+/// The test is ignored by default but runs in CI. To run the test locally,
+/// either set `RUSTFLAGS=--cfg=icu4x_check_sizes="true"` or run `cargo test`
+/// with `-- --include-ignored`.
+macro_rules! check_size {
+    ($ty:ty, $id:ident, $sizes:pat) => {
+        macro_rules! $id {
+            () => {
+                concat!(
+                    "📏 This item has a stack size of <b>",
+                    stringify!($sizes),
+                    " bytes</b> in the default ICU4X Rust toolchain."
```

Comment:
Note: I wanted to print the actual toolchain number here, but (1) Rust doesn't give a nice Rust version in an environment variable and (2) even if it did, that variable may or may not be the correct version when docs.rs is generating docs. I did manage to get the rust-toolchain.toml Rust version into an environment variable with this code in Cargo Make:

```toml
ICU4X_STABLE_TOOLCHAIN = { script = ["sed -E -n 's/^channel = \"([^\"]*)\"/\\1/p' rust-toolchain.toml"] }
```

However, if I use `env!("ICU4X_STABLE_TOOLCHAIN")`, then the macro does not compile unless that environment variable is set, which seems too restrictive.

Reply:
We can also hardcode the version number and add a comment to `rust-toolchain.toml`  saying to update this macro. This is fine though